napi_basics.md

```diff
@@ -0,0 +1,120 @@
+# N-API - Basics
+
+N-API is an API exposed by Node to developers that allow them to author Native add-ons 
+in an ABI stable manner. This means that these add-ons can work against future versions
+of Node without recompilation, and developers get the additional benefit that their 
+add-ons can work independent of the underlying JavaScript Virtual Machine being used 
+by Node.
+
+APIs exposed in the N-API surface are generally used to create and manipulate JavaScript
+objects. Concepts and operations generally map to ideas specified in the ECMA262 Language
+SPecification. The APIs generally have the following properties:
+- All N-API APIs generally return a status code of type `napi_status`. This status 
+can be used to determine whether the API call succeeded or failed.
+- The actual value being returned by the API is returned in an out parameter.
+- All JavaScript objects are abstracted behind an opaque type named `napi_value`.
+- In case of an error status code, additional information can be obtained using 
+`napi_get_last_error_info`
+
+## N-API types
+
+N-API exposes the following fundamental datatypes as abstractions that are consumed by 
+the various APIs. These APIs should be treated as opaque types that can be introspected
+only with other N-API APIs.
+
+### *napi_env*
+
+`napi_env` is used to represent a context that the underlying N-API implementation can use 
+to persist VM specific state.
+<< REVIEW: What else can we say here? >>
+
+### *napi_value*
+This is an opaque pointer that is used to represent a JavaScript object.
+
+### *napi_ref*
+This is the abstraction to use to reference a `napi_value`. This allows for users 
+to manage the lifetimes of JavaScript objects, including defining their minimum
+lifetimes explicitly, and also being notified when they are garbage-collected.
+
+### *napi_handle_scope*
+This is an abstraction used to provide hints to the garbage collector about the location of 
+handles to GC pointers. napi_handle_scope allows the user to provide a variable stack
+separate from the thread stack, in a manner that is efficient for scan in case 
+the underlying VM uses a garbage collector with precise-semantics.
+N-API values are created within the context of a handle scope- if the user does not 
+create a handle scope, they are created within the default handle scope.
```

Comment:
Good point @kkoopa. I'll update the doc to call this case out- however, @jasongin and I were talking about this offline- we were wondering if this could be handled at the N-API layer or whether that would be too complex.